src/kitsy-script-toolkit.js

```diff
@@ -3,271 +3,153 @@
+			if (i === functions.length) {
+				return;
 			}
-			globals.injectsDone = true;
-
-			// Rewrite scripts and hook everything up.
-			doInjects();
-			applyAllHooks();
-
-			// Start the game
-			globals.startExportedGame.apply(this, arguments);
-		};
-	}
-
-	function doInjects() {
-		queuedInjectScripts.forEach(function (injectScript) {
-			_inject(injectScript.searchString, injectScript.codeFragments);
-		});
-		_reinitEngine();
-	}
-
-	function applyAllHooks() {
-		var allHooks = new Set(Object.keys(queuedBeforeScripts).concat(Object.keys(queuedAfterScripts)));
-		allHooks.forEach(applyHook);
-	}
-
-	function applyHook(functionName) {
-		var superFn = globals[functionName];
-		var superFnLength = superFn.length;
-		var functions = [];
-		// start with befores
-		functions = functions.concat(queuedBeforeScripts[functionName] || []);
-		// then original
-		functions.push(superFn);
-		// then afters
-		functions = functions.concat(queuedAfterScripts[functionName] || []);
-
-		// overwrite original with one which will call each in order
-		globals[functionName] = function () {
-			var args = [].slice.call(arguments);
-			var i = 0;
-			runBefore.apply(this, arguments);
-
-			// Iterate thru sync & async functions. Run each, finally run original.
-			function runBefore() {
-				// All outta functions? Finish
-				if (i === functions.length) {
-					return;
-				}
 
-				// Update args if provided.
-				if (arguments.length > 0) {
-					args = [].slice.call(arguments);
-				}
-
-				if (functions[i].length > superFnLength) {
-					// Assume funcs that accept more args than the original are
-					// async and accept a callback as an additional argument.
-					functions[i++].apply(this, args.concat(runBefore.bind(this)));
-				} else {
-					// run synchronously
-					var newArgs = functions[i++].apply(this, args) || args;
-					runBefore.apply(this, newArgs);
-				}
+			// Update args if provided.
+			if (arguments.length > 0) {
+				args = [].slice.call(arguments);
 			}
-		};
-	}
 
-	function _inject(searchString, codeToInject) {
-		var args = [].slice.call(arguments);
-		codeToInject = _flatten(args.slice(1)).join('');
-
-		// find the relevant script tag
-		var scriptTags = document.getElementsByTagName('script');
-		var scriptTag;
-		var code;
-		for (var i = 0; i < scriptTags.length; ++i) {
-			scriptTag = scriptTags[i];
-			var matchesSearch = scriptTag.textContent.indexOf(searchString) !== -1;
-			var isCurrentScript = scriptTag === document.currentScript;
-			if (matchesSearch && !isCurrentScript) {
-				code = scriptTag.textContent;
-				break;
+			if (functions[i].length > superFnLength) {
+				// Assume funcs that accept more args than the original are
+				// async and accept a callback as an additional argument.
+				functions[i++].apply(this, args.concat(runBefore.bind(this)));
+			} else {
+				// run synchronously
+				var newArgs = functions[i++].apply(this, args) || args;
+				runBefore.apply(this, newArgs);
```

Comment:
Are you interested in putting the `setTimeout` around this branch's `runBefore` to make everything async?

Reply:
I considered that some more after our discord convo, and I don't think there's actually benefit to it now that the sequence of everything is handled in one place; I think all it would do is introduce more potential complications